cabal-install/Distribution/Client/Setup.hs

```diff
@@ -1140,9 +1140,14 @@ outdatedCommand = CommandUI {
      trueArg
 
     ,option [] ["new-freeze-file"]
-     "Act on the new-style freeze file"
-     outdatedNewFreezeFile (\v flags -> flags { outdatedNewFreezeFile = v })
-     trueArg
+     "Act on the new-style freeze file named cabal.project.freeze"
+     outdatedProjectFile (\_ flags -> flags { outdatedProjectFile = pure "cabal.project" })
+     (noArg mempty)
+
+    ,option [] ["project-file"]
```

Comment:
Something doesn't smell right to me about duplicating the global `--project-file=` flag as a local `outdated` sub-command flag. What does this make possible that wasn't possible before?

Reply:
I'm sorry, I don't understand this question. Where is the global `--project-file` flag being defined? `outdated` doesn't currently accept it:

```
$ cabal outdated --project-file=foo.project
cabal: unrecognized 'outdated' option `--project-file=foo.project'
$ cabal --version
cabal-install version 2.2.0.0
compiled using version 2.2.0.1 of the Cabal library 
```

Bear in mind that `outdated` isn't really a new-style command: it's more like a v1 command that happens to know a little bit about new-style freeze files. `new-outdated` as outlined in #4831 might be a worthy project, but that's a much bigger task. I think that this patch has a good power-to-weight ratio.

Reply:
Oh you're right, I stand corrected. I confused `--project-file=` to be a global one (like `--store-dir=`)

Now this is starting to make more sense. But if we're going to migrate to make the `new-*`-style commands default, then it's also kinda strange to have `--project-file=` imply a side-effect; rather than have it act like merely setting a property, and requiring an explicit `--new-freeze-file`; that would feel more consistent to me w/ how `--project-file=` usually acts.

Reply:
I've gone ahead and implemented that. In the future we'll probably want to remove `--new-freeze-file` outright; I've made sure that `--project-file` without it errors out, so that we've got freedom to make that change.